internal/cli/cmd/install.go

```diff
@@ -96,6 +101,9 @@ cilium install --context kind-cluster1 --cluster-id 1 --cluster-name cluster1
 	cmd.Flags().StringVar(&params.Azure.ClientID, "azure-client-id", "", "Client (application) ID of Azure Service Principal to use for installing Cilium (will create one if none provided)")
 	// It can be deprecated since we have a helm option for it
 	cmd.Flags().StringVar(&params.Azure.ClientSecret, "azure-client-secret", "", "Client secret of Azure Service Principal to use for installing Cilium (will create one if none provided)")
+	cmd.Flags().BoolVar(&params.Azure.IsBYOCNI, "azure-force-byocni", false, "Explicitly state we're using an AKS cluster in BYOCNI mode. Bypasses all AKS auto-detection if provided logic, removing the need for any other Azure flag.")
```

Comment:
:thinking: why isn't `--datapath-mode=azure-byocni` enough?

Reply:
^ It seems to me that the only reason the autodetect logic is there is for `--datapath-mode=azure` where it needs to set up the service principal, otherwise we could just skip the azure autodetect logic altogether whien `--datapath-mode=azure-byocni`...?

Reply:
^ idea, just throwing it out there:

* If datapath-mode is empty, autodetect datapath-mode (set to azure-byocni)
* Then, if datapath-mode is azure, autodetect azure settings; do not do this for azure-byocni

Reply:
> ^ It seems to me that the only reason the autodetect logic is there is for `--datapath-mode=azure` where it needs to set up the service principal, otherwise we could just skip the azure autodetect logic altogether whien `--datapath-mode=azure-byocni`...?

The auto-detection mechanism requires `--azure-resource-group` to be passed in so that we can run `az aks show` and retrieve cluster info to determine if the cluster has been created in BYOCNI mode or not (see `azureRetrieveAKSClusterInfo` in `azure.go`), so we can't just skip the auto-detection logic when nothing is passed in, however you are right that when it's explicitly set all of this can be skipped. I just didn't think of it 🤦🏻